File: erpcloud_itagksa/itag_manufacturing/material_request/material_request.js
frappe.ui.form.on("Material Request", {
    refresh(frm) {
        frappe.itagksa.set_collab_serial_query(frm);
        prefill_supplier_store(frm);
    },
    custom_is_collaboration_service_po(frm) {
        prefill_supplier_store(frm);
    },
});

frappe.ui.form.on("Material Request Item", {
    custom_serial_no(frm, cdt, cdn) {
        frappe.itagksa.on_collab_serial_set(frm, cdt, cdn);
    },
});

function prefill_supplier_store(frm) {
    if (!frm.doc.custom_is_collaboration_service_po || frm.doc.custom_supplier_store) {
        return;
    }
    frappe.db.get_single_value("ITAG KSA Settings", "default_supplier_store").then((store) => {
        if (store && !frm.doc.custom_supplier_store) {
            frm.set_value("custom_supplier_store", store);
        }
    });
}

File: erpcloud_itagksa/itag_manufacturing/material_request/material_request.py
# Copyright (c) 2026, Globcom Qatar and contributors
# For license information, please see license.txt

"""
Material Request collaboration hook for ITAG KSA.

validate: When MR is a collaboration service request, ensure each row's Serial
No, if set, belongs to the row's custom_sub_item Item and is not already
delivered.
"""

from frappe import _

from erpcloud_itagksa.itag_manufacturing.utils.collab_serial import validate_serial_no_items
from erpcloud_itagksa.itag_manufacturing.utils.supplier_store import apply_default_supplier_store


def validate(doc, method=None):
    apply_default_supplier_store(doc)
    if not doc.get("custom_is_collaboration_service_po"):
        return
    validate_serial_no_items(doc, _("Collaboration MR validation"))
